pkg/ddl/partition.go

```diff
@@ -97,7 +97,7 @@ func checkAddPartition(t *meta.Mutator, job *model.Job) (*model.TableInfo, *mode
 func (w *worker) onAddTablePartition(jobCtx *jobContext, job *model.Job) (ver int64, _ error) {
 	// Handle the rolling back job
 	if job.IsRollingback() {
-		ver, err := w.onDropTablePartition(jobCtx, job)
+		ver, err := w.rollbackLikeDropPartition(jobCtx, job)
```

Comment:
As DDL execution state machine the rollbackone statement still need go through some state to Public that does in onDropTablePartition func does. In rollbackLikeDropPartition I did not see this kind of state machine path execution?

Reply:
The rollback logic has not been changed in this PR, just moved the code to a separat function.

For ADD partition, there is only two states, so the first state transition should cancel directly, without need for rollback, the second state transition it can do rollback directly in a single state transition.

For REORGANIZE partition, it will normally be called after `convertReorgPartitionJob2RollbackJob()`, which does handle state transitions where needed, to then allow `rollbackLikeDropPartition` to do the final cleanup.